changelog.d/9247.feature

```diff
@@ -0,0 +1 @@
+Add experimental support for backfilling history into rooms ([MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716)).
```

Comment:
Do we not need to change the behaviour of getting backfill from the DB? Or are we punting that to a separate PR?

Reply:
I'm going to work on the insertion/marker event in another PR after this merges so it doesn't grow more and I don't have to keep refactoring across PR's.

Reply:
SGTM!